base/gocb_utils.go

```diff
@@ -153,3 +158,52 @@ func getRootCAs(caCertPath string) (*x509.CertPool, error) {
 	}
 	return rootCAs, nil
 }
+
+// waitForGoCB response waits for a response from a gocb.Agent
+func waitForGoCBResponse(signal <-chan error, op gocbcore.PendingOp, timeout time.Duration) error {
+	timeoutTmr := gocbcore.AcquireTimer(timeout)
+	select {
+	case err := <-signal:
+		gocbcore.ReleaseTimer(timeoutTmr, false)
+		return err
+	case <-timeoutTmr.C:
+		gocbcore.ReleaseTimer(timeoutTmr, true)
+		WarnfCtx(context.TODO(), "gocbcore: Request has timed out, canceling op")
+		if op != nil {
+			op.Cancel()
+			// wait for confirmation after canceling the PendingOp
+			<-signal
+		}
+		return gocbcore.ErrTimeout
+	}
+}
+
+func getCollectionIDs(scope, collection string, agent *gocbcore.Agent) ([]uint32, error) {
```

Comment:
I think you could design this now to work against a slice of collections, even if we're only invoking it for a single collection today.